### PR TITLE
adding dashboard/config/courses/*.course to sync-in

### PR DESCRIPTION
**Pull request by mgc1194**
This PR tackles a series of Jira tickets regarding CSD translatability bugs.

Currently the [landing page](https://studio.code.org/courses/csd-2022) for the CSD course have a couple items that are not translated.
This elements are translatable but the content has not been added to the sync.
In PR [#50450](https://github.com/code-dot-org/code-dot-org/pull/50450) `dashboard/config/locales/courses.en.yml` was added to the sync.
However some content of teacher resources dropdown menu was not translated yet.
image

The explanation for this partial translatability bug resides in resources being synced from `scrip_json` files. These files store content from Units, such as `csd1-2022.script_json`. In addition, resources for the whole course are stored in the `dashboard/config/courses` directory. These `*.course` files are not synced, therefore the only resources that appear translated are those shared between whole course and individual units.

This PR adds to the sync-in CSD resources stored in `*.course` files. Morover, it stores the added resources ussing the same key structure as Unit resources, that way, the existing i18n functions will be able to find this resources without any further changes.

## Links

- jira ticket: [P20-37](https://codedotorg.atlassian.net/browse/P20-37)
- jira ticket: [P20-38](https://codedotorg.atlassian.net/browse/P20-38)
- jira ticket: [P20-45](https://codedotorg.atlassian.net/browse/P20-45)

## Testing story

I ran the full sync:
- Checked that resources were correctly added to `courses.yml`
- Checked on crowding that resources were uploaded and created custom translations in the testing Crowdin project.
- Checked that content was correctly distributed

Screenshot 2023-03-08 at 13 12 37

## Deployment strategy
Changes made by this PR will be effective after the next i18n sync.

## Follow-up work

Some resources are not available for localization intentionally.

This PR maintains yml files as localization files. These files should be transitioned into json files as part of tech-debt reduction work.

## Security

As always, allowing translators to modify links is a potential risk.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/sync-in.rb [hunk 1]
```diff
@@ -556,6 +557,40 @@ def write_to_yml(type, strings)
...
+# https://studio.code.org/courses/csd-2021
+# All other resources used in Unit come from scrip_json files (used in the landing pages for each Unit).
+# https://studio.code.org/s/csd1-2021
+def localize_course_resources
+  puts "Preparing course resources"
+
+  # Currently, csd is the only fully translatable course that has resources in this directory
+  translatable_courses = %w(csd)
+  resources = {}
+  Dir.glob(File.join('dashboard', 'config', 'courses', '*.course')).sort.each do |file|
```
Comment (nicklathe) [inline bin/i18n/sync-in.rb]: Do you need `.sort` here? Aren't you just iterating over all of the `*.course` files in `dashboard/config/courses`?

Comment (mgc1194) [reply bin/i18n/sync-in.rb]: I don't think it is necesary

Comment (nicklathe) [reply bin/i18n/sync-in.rb]: I agree 👍. Using sort here just sorts the array before you iterate over it, only adding time complexity.

**Comment by nicklathe** [comment]
This looks good to me. I'd say we should add unit tests for these new functions, but it doesn't look like we have any for the i18n stuff anyway, unless I'm looking in the wrong spot.

**Review by nicklathe — APPROVED** [review]
LGTM. I left another comment about missing unit tests for these new functions, however it doesn't look like we have many (if any) existing unit tests for the i18n pipeline, so that should be addressed outside of this PR.

**Pull request merged**